presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/TestRun.java

```diff
@@ -168,21 +175,32 @@ public void run()
 
         private Environment tryStartEnvironment()
         {
-            log.info("Pruning old environment(s)");
-            Environments.pruneEnvironment();
-
-            log.info("Creating environment '%s' with configuration %s", environment, environmentConfig);
             Environment environment = getEnvironment();
-            log.info("Starting the environment '%s'", environment);
-            environment.start();
+
+            if (!attach) {
+                log.info("Pruning old environment(s)");
+                Environments.pruneEnvironment();
+
+                log.info("Starting the environment '%s' with configuration %s", this.environment, environmentConfig);
+                environment.start();
+            }
+            else {
+                DockerContainer tests = (DockerContainer) environment.getContainer("tests");
+                tests.clearDependencies();
+                tests.setNetwork(new ExistingNetwork(Environment.PRODUCT_TEST_LAUNCHER_NETWORK));
```

Comment:
you can also enumerate networks using DockerClient: `DockerClientFactory.instance().client().listNetworksCmd().withIdFilter(PRODUCT_TEST_LAUNCHER_NETWORK).exec()`

Reply:
That way you will be sure that network exists

Reply:
This adds validation (partial - network may exist but nothing else), but i won't free me from having `ExistingNetwork` class.
I need `org.testcontainers.containers.Network` and not `com.github.dockerjava.api.model.Network`

i am tempted to leave as is